File: 2022/220818-dojo/javascript/solution.js
// eslint-disable-next-line import/prefer-default-export
export function solution(s) {
  const answer = s.split(' ')
    .map((word) => word.split('')
      .map((value, i) => (i % 2 === 0 ? value.toUpperCase() : value.toLowerCase()))
      .join(''))
    .join(' ');
  return answer;
}

File: 2022/220818-dojo/java/app/src/main/java/Solution.java
public class Solution {
  public String solution(String s) {
    String answer = "";

    String[] words = s.split("");

    int index = 0;

    for (int i = 0; i < words.length; i += 1) {
      if (words[i].equals(" ")) {
        index = 0;
        continue;
      }

      if (index % 2 == 0) {
        words[i] = words[i].toUpperCase();
      }

      if (index % 2 != 0) {
        words[i] = words[i].toLowerCase();
      }

      index += 1;
    }

    answer = String.join("", words);
    return answer;
  }
}
